source/module_base/matrix.cpp

```diff
@@ -321,33 +323,34 @@ double matrix::trace_on(void) const
     return tr;
 }
 
-void matrix::get_extreme_eigen_values(double &ev_lower, double &ev_upper)const
-{
-    double *a = new double[nr];
-    double *b = new double[nr];
-    for (int i = 0; i < nr; ++i)
-    {
-        double sum = 0.0;
-        for(int j = 0; j < nc; ++j)
-        {
-            sum += fabs(c[i * nc + j]);
-        }
-        sum -= fabs(c[i * nc + i]);
-        a[i] = c[i * nc + i] - sum;
-        b[i] = c[i * nc + i] + sum;
-    }
-
-    ev_lower = a[0];
-    ev_upper = b[0];
-
-    for (int i = 1; i < nr; ++i)
-    {
-        if (a[i] < ev_lower) ev_lower = a[i];
-        if (b[i] > ev_upper) ev_upper = b[i];
-    }
-    delete[] a;
-    delete[] b;
-}
+//this function is wrong, and there is no file use this function.
```

Comment:
Why is it wrong?

Reply:
The name of the function is void matrix::get_extreme_eigen_values,which means that this function should return the maximum and minimum of the eigenvalues of this matrix. However, firstly it has no insurance code to make sure the matrix is a square matrix. Secondly, there is no code to solve the eigenvalues of the matrix. This function is right if and only if the matrix has been diagonalized. The same problem is that there is no insurance code and it's not the most efficient algorithm.
In conclusion, this function can't perfectly finish the aim defined by its name. And it has not been called.
https://hongritianqi.github.io/abacus-develop/source/html/classModuleBase_1_1matrix.html#a976ec94fbdf0a34ae89c92cdcfb9ddc8